### DISCUSSION TITLE
Lesson 11: Imported NFTs are not showed correctly in my MetaMask

### DISCUSSION THREAD

**Discussion by NanaChiki** [Cyfrin/foundry-full-course-cu#1054] [Q&A] [answered]
I've been trying to see the NFTs in my MetaMask, but it 's continuously showed something like this.I tried to do it with the same uri of this repo, or creating a JSON file with the PUG imageURI but I still got the same issue.

And I almost read all discussions relative on it, but I've not found some solutions out.
I would really appreciate if there're someone know how to fix it and give me some tips!
Thanks!

<img width="1333" alt="Screenshot 2023-10-27 at 12 15 32 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/138076936/7de48e8f-7d65-49ea-af9e-bd98210936a1">

![Screenshot 2023-10-27 at 11 16 13 AM](https://github.com/Cyfrin/foundry-full-course-f23/assets/138076936/70035adb-4c8b-423f-a403-ab54a97c2475)

**Comment by Niferu** [comment]
Hi @NanaChiki 

I could be wrong but I would remove this part `?filename=pug.png` from your metadata. I can tell you that I'm posting NFT's with following "image" attribute in metadata and those are working perfectly fine: "https://ipfs.io/ipfs/QmeSRVQ3LnfPfy78UxoNFCJVLeFjyxmgR3wotESV5ziYCD"

So just edit your script to remove `?filename=pug.png` while creating metadata and you should be good.

**Reply by NanaChiki** [reply]
Thanks! @Neftyr 

I'd just try to change my image attribute in metadata to "https://ipfs.io/ipfs/QmYx6GsYAKnNzZ9A6NvEKV9nf1VaDzJrqDR23Y8YSkebLU", but still nothing change...😭
What makes me different from everyone else? 🤔

![Screenshot 2023-10-28 at 8 16 53 AM](https://github.com/Cyfrin/foundry-full-course-f23/assets/138076936/c294d9b8-9545-4521-a757-19e0e23f7a0d)

Here are my new deployed contract and tokenURI:
contract: 0x6c27852a34c817706f18Bbb8a31715433BA3B5b2
tokenURI: ipfs://bafybeig3t34gaz7el4uf4nxenect33nuany6bfo2hkuxvpqwc7lxbztygy

**Reply by Niferu** [reply]
Hi @NanaChiki 

Hmm it is definitely something wrong with your image as I cannot see it in browser. My point was that you should be using tokenURI which is equal to ipfs link (one that everyone can access in their browser) to metadata and in that metadata we have another link which is under attribute "image" and that link should also be working in your browser. You copy metadata link (tokenURI) to your browser you see metadata in browser, you copy image link from that metadata to your browser -> you see image without changing anything and without installing any additional plugins, thats why I showed you my ipfs link format, your clearly do not lead anywheere that is why you do not see image in metamask too it is logical right?

**Reply by NanaChiki** [reply]
Hi @Neftyr 

Thanks to you, apparently, I realized that I was using the wrong tokenURI.
Can you see each of these ipfs supported tokenURI and https supported tokenURI bellow in your browser btw?

✅TokenURI
IPFS: ipfs.io/ipfs/QmWp7ZkfCBDvDr9LjmQk4F95gZeS2cbrS2Pr8gbDH6aCmZ
Https: https://ipfs.io/ipfs/QmWp7ZkfCBDvDr9LjmQk4F95gZeS2cbrS2Pr8gbDH6aCmZ

✅ImageURI
IPFS: ipfs.io/ipfs/QmYx6GsYAKnNzZ9A6NvEKV9nf1VaDzJrqDR23Y8YSkebLU
Https: https://ipfs.io/ipfs/QmYx6GsYAKnNzZ9A6NvEKV9nf1VaDzJrqDR23Y8YSkebLU

I think these URIs would be a correct format as you mentioned...

**Reply by Niferu** [reply]
Hi @NanaChiki 

ImageURI works perfectly fine for me now, so if you will use the same in your metadata it should be correctly displayed in your metamask :)

**Reply by NanaChiki** [reply]
Hi @Neftyr 

I got the same when I'd minted a NFT based on base64 encoded tokenURI, so I suspect it may not be a problem coming from IPFS, but a storage problem with my computer that processes the images 

I don't know how to resolve this, so I'll move on for now, thanks for taking the time

**Reply by Niferu** [reply]
Hi @NanaChiki 

It is definitely not problem with storage on your computer. If you minted NFT correctly you should be able to see it on open sea for example, just change network to sepolia or whatever you have created it on. If URI is correct you should be able to see image on OpenSea if not, then you have minted NFT with incorrect tokenURI and you will not see it on metamask either.

**Reply by ikennaumeh** [reply, marked as answer]
Go to meta mask, go to settings -->> advanced --- >> clear meta mask logs.
Then go to security and privacy, enable autoDetectNft and openSea api

**Reply by NanaChiki** [reply]
@Nlferu, @ikennaumeh 

It was displayed successfully! You guys are my heroes, thank you so much!!!!!
![Screenshot 2023-11-03 at 10 55 31 AM](https://github.com/Cyfrin/foundry-full-course-f23/assets/138076936/f0643e2c-f039-429f-b9cc-565e59a1023f)

**Reply by ikennaumeh** [reply]
Uwc

**Reply by Osneue** [reply]
Hi @NanaChiki
I encountered the same problem.
And I try the same solution, but it doesn't work for me.
it's normal on OpenSea, it's really strange.
I try the following steps, is it right?
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/143680960/6c836c84-a90d-4c3c-838b-e8d4468b9b1f)
and
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/143680960/f162a79f-3eef-4020-bb9a-67e53381c492)
I reimport my nft. Nothing has changed.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/143680960/951142e8-814f-4efc-97b1-952c77fafc46)

**Reply by NanaChiki** [reply]
Hi @Osneue

Yeah, the step you have taken are correct.Perhaps, tokenURI or ImageURI you used was incorrect, so try once using ImageURI, and tokenURI copied from the "Share link", as follows.


 
<img width="835" alt="Screenshot 2023-11-06 at 11 44 12 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/138076936/26a54f90-d1e9-4035-b722-01e965f18ef6">


<img width="1310" alt="Screenshot 2023-11-06 at 11 44 31 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/138076936/b3e39e39-58a9-471f-93f3-8f007c0b5ae3">

**Comment by Amio13** [comment]
Any solution for _exist at the TokenURI function ? In the lesson 11?

**Reply by NanaChiki** [reply]
Hi @Amio13 
Could you go into detail on that matter??

**Comment by Blockchain-Oracle** [comment]
I think the issue is from meta mask , try deploying on sepolia testnet and confirm your NFT ON 
[OPEN  SEA SEPOLIA](https://testnets.opensea.io/)

**Comment by Amio13** [comment]
ok thanks
